<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FlashProg</class>
 <widget class="QMainWindow" name="FlashProg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>820</width>
    <height>592</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>820</width>
    <height>550</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>29 series flash programmer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>662</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="open">
       <property name="text">
        <string>Open</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="read">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Read</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="write">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Write</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="erase">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Erase</string>
       </property>
       <property name="icon">
        <iconset theme="Erase">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="autoerase">
       <property name="text">
        <string>Autoerase </string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="autoverify">
       <property name="text">
        <string>Verify</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>197</width>
      <height>94</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QComboBox" name="com"/>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QPushButton" name="update">
         <property name="text">
          <string>Update</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="conn">
         <property name="text">
          <string>Connect</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QComboBox" name="selectChip"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>500</y>
      <width>741</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Â© Xiorcid, 2024</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>                            State:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="state">
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>Not Connected</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QProgressBar" name="progressBar">
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QTextBrowser" name="filecontent">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>30</y>
      <width>571</width>
      <height>461</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Cascadia Mono</family>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="flashimg">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>142</y>
      <width>181</width>
      <height>351</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>820</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="geometry">
     <rect>
      <x>269</x>
      <y>153</y>
      <width>174</width>
      <height>168</height>
     </rect>
    </property>
    <property name="contextMenuPolicy">
     <enum>Qt::DefaultContextMenu</enum>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuRecent">
     <property name="title">
      <string>Recent</string>
     </property>
    </widget>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="menuRecent"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <widget class="QMenu" name="menuChip">
     <property name="title">
      <string>Chip</string>
     </property>
     <addaction name="separator"/>
    </widget>
    <widget class="QMenu" name="menuUART_speed">
     <property name="title">
      <string>UART speed</string>
     </property>
     <addaction name="separator"/>
    </widget>
    <addaction name="actionAutoerase"/>
    <addaction name="actionAutoconnect"/>
    <addaction name="menuChip"/>
    <addaction name="menuUART_speed"/>
    <addaction name="separator"/>
    <addaction name="actionSave_settings"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSettings"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionOpen_file">
   <property name="text">
    <string>Open file</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open       Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Save         Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit           Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAutoerase">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Autoerase</string>
   </property>
  </action>
  <action name="actionSave_settings">
   <property name="text">
    <string>Save settings</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>Help          F1</string>
   </property>
  </action>
  <action name="actionAutoconnect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Autoconnect</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
